src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -544,30 +556,98 @@ private Boolean isPassable(final BlockState blockState)
         return blockState.getMaterial() == Material.LEAVES;
     }
 
+
     /**
      * Check if distance to block changed and if we are not moving for too long, try to get unstuck.
      *
      * @return false
      */
-    private boolean checkIfStuckOnLeaves()
+    private boolean checkIfStuck()
     {
         if (!worker.getNavigator().noPath())
         {
-            Path pathToTree = worker.getNavigator().getPath();
-            if (pathToTree != null && pathToTree.getCurrentPathLength() > pathToTree.getCurrentPathIndex())
+            Path path = worker.getNavigator().getPath();
+            if (path != null)
             {
-                PathPoint next = pathToTree.getPathPointFromIndex(pathToTree.getCurrentPathIndex());
-                BlockPos nextPos = new BlockPos(next.x, next.y, next.z);
-                BlockPos nextPosUp = new BlockPos(next.x, next.y + 1, next.z);
-
-                if (world.getBlockState(nextPos).getBlock().isIn(BlockTags.LEAVES))
+                if (path.getCurrentPathLength() > path.getCurrentPathIndex())
+                {
+                    return true;
+                }
+                if (path.getCurrentPathLength() == 0)
                 {
-                    mineBlock(nextPos);
+                    return true;
                 }
-                else if (world.getBlockState(nextPosUp).getBlock().isIn(BlockTags.LEAVES))
+            }
+            else
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Mines leaves on path and above.
+     */
+    private void tryUnstuck()
+    {
+        if (!worker.getNavigator().noPath())
+        {
+            Path path = worker.getNavigator().getPath();
+            if (path != null)
+            {
+                // Unstuck with path
+                ArrayList<BlockPos> checkPositions = new ArrayList<>();
+                PathPoint next = path.getPathPointFromIndex(path.getCurrentPathIndex());
+
+                // Blocks in front of the worker
+                for(int i = 0; i <= 2; i++)
                 {
-                    mineBlock(nextPosUp);
+                    checkPositions.add(new BlockPos(next.x, next.y + i, next.z));
                 }
+
+                // Block above the worker
+                checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 2, worker.getCurrentPosition().getZ()));
+
+                mineIfEqualsBlockTag(checkPositions, BlockTags.LEAVES);
+                return;
+            }
+        }
+
+        // General unstuck
+        ArrayList<BlockPos> checkPositions = new ArrayList<>();
+        List<Direction> directions = new ArrayList<>();
+        directions.add(Direction.NORTH);
+        directions.add(Direction.EAST);
+        directions.add(Direction.WEST);
+        directions.add(Direction.SOUTH);
+
+        for (Direction direction: directions)
```

Comment:
There is a minecraft class for all horizontal directions =D

Reply:
Couldn't find it :(

Reply:
Plane.HORIZONTAL.values()

Reply:
Is done :)